File: backend/src/main/java/es/ual/dra/agrotrack/dto/response/ProductoResponse.java
package es.ual.dra.agrotrack.dto.response;

import es.ual.dra.agrotrack.model.entity.Producto;

public record ProductoResponse(
    Long id,
    String nombre,
    String descripcion,
    String imagenUrl,
    Integer temporadaInicio,
    Integer temporadaFin,
    CategoriaResponse categoria
) {
    public static ProductoResponse from(Producto p) {
        return new ProductoResponse(
            p.getId(),
            p.getNombre(),
            p.getDescripcion(),
            p.getImagenUrl(),
            p.getTemporadaInicio(),
            p.getTemporadaFin(),
            CategoriaResponse.from(p.getCategoria())
        );
    }
}

File: frontend/src/app/models/response/producto-response.ts
import { CategoriaResponse } from './categoria-response';

export interface ProductoResponse {
  id: number;
  nombre: string;
  descripcion: string | null;
  imagenUrl: string | null;
  temporadaInicio: number | null;
  temporadaFin: number | null;
  categoria: CategoriaResponse;
}
